controllers/database_reconciler_helpers.go

```diff
@@ -253,3 +258,124 @@ func (r *DatabaseReconciler) handleSync(ctx context.Context, database *ndbv1alph
+	err = r.Get(ctx, namespacedName, foundService)
+	if err != nil && errors.IsNotFound(err) {
+		log.Info("No service found, creating a new service", "target port", targetPort)
+		service := &corev1.Service{
+			ObjectMeta: metadata,
+			Spec: corev1.ServiceSpec{
+				Ports: []corev1.ServicePort{
+					{
+						Protocol:   corev1.ProtocolTCP,
+						Port:       80,
+						TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: targetPort},
+					},
+				},
+			},
+		}
+		// Setting database as the owner of this service
+		err = ctrl.SetControllerReference(database, service, r.Scheme)
+		if err != nil {
+			log.Error(err, "Error setting controller reference for the service")
+		}
+		err = r.Create(ctx, service)
+		if err != nil {
+			log.Error(err, "Failed to create a new service")
+			return
+		}
+		log.Info("Created a new service", "service name", service.GetName())
+	}
+	log.Info("Returning from database_reconciler_helpers.setupService")
+	return
+}
+
+// Checks and creates an endpoints object for the service if it does not already exists.
+// If it is already present, syncs the IP address with the Database.Status.IPAddress if out of sync.
+func (r *DatabaseReconciler) setupEndpoints(ctx context.Context, database *ndbv1alpha1.Database, namespacedName types.NamespacedName, metadata metav1.ObjectMeta, targetPort int32) (err error) {
+	log := ctrllog.FromContext(ctx)
+	log.Info("Entered database_reconciler_helpers.setupEndpoints")
+	foundEndpoint := &corev1.Endpoints{}
+	endpointSubsets := []corev1.EndpointSubset{
+		{
+			Addresses: []corev1.EndpointAddress{{IP: database.Status.IPAddress}},
+			Ports:     []corev1.EndpointPort{{Port: targetPort}},
+		},
+	}
+	err = r.Get(ctx, namespacedName, foundEndpoint)
+	// Create an endpoint if it does not exists.
+	if err != nil && errors.IsNotFound(err) {
+		log.Info("No endpoint found, creating a new endpoint")
+		endpoint := &corev1.Endpoints{
+			ObjectMeta: metadata,
+			Subsets:    endpointSubsets,
+		}
+		// Setting database as the owner of this endpoint
+		ctrl.SetControllerReference(database, endpoint, r.Scheme)
+		err = r.Create(ctx, endpoint)
+		if err != nil {
+			log.Error(err, "Failed to create a new ep")
+			return
+		}
+		log.Info("Created a new endpoint", "endpoint name", endpoint.GetName())
+	} else {
+		// If endpoint exists, check if the IP has changed.
+		// If changed, sync with the latest IP in the database CR status.
+		for _, subset := range foundEndpoint.Subsets {
+			for _, address := range subset.Addresses {
+				if address.IP == database.Status.IPAddress {
+					// IP has not changed, no need to update endpoint
+					return
+				}
+			}
+		}
+		log.Info("Endpoint found with a different IP address, updating.")
+		foundEndpoint.Subsets = endpointSubsets
+		err = r.Update(ctx, foundEndpoint)
+		if err != nil {
+			log.Error(err, "Failed to update endpoint")
+			return
+		}
+	}
+	log.Info("Returning from database_reconciler_helpers.setupEndpoints")
```

Comment:
Rather than this type of comments, would it make sense to say that you are were able to successfully were able to set endpoints for a given database, under a given namespace, etc

Reply:
Ohh! What I'm trying to convey is that it would always be better to add logical log statements rather than saying that we are entering method x and exiting method x. This works really well when we are trying to debug. But, adding log statements would make more sense.

Reply:
Yes, they are printed **only** when it is created.
![image](https://user-images.githubusercontent.com/107612534/190508014-bc96cc2b-6bd4-4360-b56e-d8c33ab9afc5.png)